Xcorr update

rh-downunder opened the pull request:
The PR contains few code changes but is the result of a months-long signal-processing pipeline optimization in xcorqc.py, for producing optimal Green's functions.

medlin01GA commented on seismic/xcorqc/xcorqc.py:
```diff
@@ -84,40 +84,44 @@ def taper(tr, taperlen):
-    l = x.shape[0]
-    h = signal.hann(l)
-    xf = rfft(h*x)
+    # smooth amplitude spectrum
```
It would be good to have a bit more explanation in the following 10 lines or so of code.

rh-downunder replied:
Just pushed some comments through

medlin01GA commented on seismic/xcorqc/xcorqc.py:
```diff
@@ -206,21 +206,15 @@ def xcorr2(tr1, tr2, sta1_inv=None, sta2_inv=None,
+                tr2_d = signal.detrend(tr2_d)
 
-                # zero-mean
+                # STEP 2: demean
```
I thought the detrending in STEP 1, which subtracts a fitted line y = a*x + b, would have removed the mean already, because b is the mean term.

rh-downunder replied:
This is Babak's special recipe :-).

rh-downunder commented:
@medlin01GA , I've also updated the Readme under xcorqc with setup instructions for Gadi. I've also updated the validation scripts you wrote earlier to test setups on Raijin. Please have a look when you get a chance.
